### DISCUSSION TITLE
Multiple BASE_URLs

### DISCUSSION THREAD

**Discussion by deployn** [plausible/analytics#1800] [Self-Hosted Support] [answered]
Hey,

I'm self-hosting Plausible, but I use multiple docker containers as I need Plausible to listen to plausible.domain1.com and plausible.domain2.com. Is there a way to use more than one BASE_URL parameter on one instance?

**Comment by jvandenaardweg** [comment, marked as answer]
If you are running nginx on that machine, you can just add additional server_name's to the plausible config and setup your DNS CNAME or A for each domain to target your main instance.

Config: https://github.com/plausible/hosting/blob/master/reverse-proxy/nginx/plausible

Example:
```
server_name example.com;
server_name example2.com;
server_name example3.com;

```

I'm currently running one plausible instance on 4 different domains, to bypass ad blockers.

The only downside is, the suggested script that the dashboard provides when creating a new site, is always the one from your main instance, as it probably uses that base config value. For me it's no issue, as i know what script path to use as it is just me looking at stats for 4 websites i own.

**Reply by deployn** [reply]
Thank you, I knew how to add multiple domains to one proxy, but I thought there could be some more problems than the suggested script.

**Reply by jvandenaardweg** [reply]
One other thing to keep in mind: your /register and /login pages are also accessible on all those domains.

I disabled signups, because I can manage that myself. You can do this by using `DISABLE_REGISTRATION=true` in the .env file for Plausible. 

I moved my other domains to a separate server block in the config and return a 404 for the login page. Like this:

```
# Main instance
server {
        server_name analytics.example.com;

        listen 80;
        listen [::]:80;

        location / {
                proxy_pass http://127.0.0.1:8000;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

}

# Other domains
server {
        server_name a.example1.com;
        server_name a.example2.com;
        server_name a.example3.com;

        listen 80;
        listen [::]:80;

        location / {
                proxy_pass http://127.0.0.1:8000;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

        # You can do the same for /register if you want
        location = /login {
                try_files $uri $uri/ =404;
        }

}
```

So I can use my main instance URL to view all the stats from the websites I own. And use `https://a.example1.com/js/script.js` to load Plausible and bypassing ad blocking scripts.

If you have strict SSL certificates, this all might be challenging to setup. My domains are just behind CloudFlare using the "Flexible" SSL option. So no need to setup certificates.
